pkg/apiserver/rest/usecase/envbinding.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package usecase
+
+import (
+	"context"
+	"errors"
+
+	"github.com/oam-dev/kubevela/pkg/apiserver/datastore"
+	"github.com/oam-dev/kubevela/pkg/apiserver/model"
+	apisv1 "github.com/oam-dev/kubevela/pkg/apiserver/rest/apis/v1"
+)
+
+// EnvBindingUsecase envbinding usecase
+type EnvBindingUsecase interface {
+	GetEnvBindings(ctx context.Context, app *model.Application) ([]*apisv1.EnvBindingBase, error)
+	CreateEnvBinding(ctx context.Context, app *model.Application, env apisv1.CreateApplicationEnvRequest) (*apisv1.EnvBinding, error)
+	BatchCreateEnvBinding(ctx context.Context, app *model.Application, env apisv1.EnvBindingList) error
+	UpdateEnvBinding(ctx context.Context, app *model.Application, envName string, diff apisv1.PutApplicationEnvRequest) (*apisv1.EnvBinding, error)
+	DeleteEnvBinding(ctx context.Context, app *model.Application, envName string) error
+	DetailEnvBinding(ctx context.Context, envBinding *model.EnvBinding) (*apisv1.DetailEnvBindingResponse, error)
+}
+
+type envBindingUsecaseImpl struct {
+	ds datastore.DataStore
+}
+
+// NewEnvBindingUsecase new envBinding usecase
+func NewEnvBindingUsecase(ds datastore.DataStore) EnvBindingUsecase {
+	return &envBindingUsecaseImpl{
+		ds: ds,
+	}
+}
+
+func (c *envBindingUsecaseImpl) CreateEnvBinding(ctx context.Context, app *model.Application, envReq apisv1.CreateApplicationEnvRequest) (*apisv1.EnvBinding, error) {
+	envBinding, err := c.getBindingByEnv(ctx, app, envReq.Name)
+	if err != nil {
+		if !errors.Is(err, datastore.ErrRecordNotExist) {
+			return nil, err
+		}
+	}
+	envBindingModel := convertCreateReqToEnvBindingModel(app, envReq)
+	if envBinding != nil {
+		if err := c.ds.Put(ctx, &envBindingModel); err != nil {
```

Comment:
No updates should be made in the creation logic.  If exist, should return `bcode.ErrorEnvbindExist`